src/context/index.ts

```diff
@@ -170,6 +171,28 @@ export class Context {
     return new Context({ polymeshApi, middlewareApi, keyring });
   }
 
+  /**
+   * Get if the current node is archive or not
+   */
+  public async isCurrentNodeArchive(): Promise<boolean> {
+    const {
+      polymeshApi: {
+        query: { balances, system },
+      },
+    } = this;
+
+    try {
+      const blockHash = await system.blockHash(0);
+      await requestAtBlock(balances.totalIssuance, {
+        args: [],
+        blockHash,
+      });
```

Comment:
Can you double check with the core team whether you actually need to call a second function? It seems like the call to `blockHash` itself should be enough to know if the node is an archive node

Reply:
Mudit told me to fetch something known and check that exact value is returned because it's not guaranteed that the node will return an error if it doesn't have the data. It might just return garbage instead an error

Reply:
I'm going to rewrite some lines here